### PR TITLE
Add for witness in snapshot meta

### PR DESCRIPTION
**Pull request by Connor1996**
Generate an empty snpashot for wintess is okay, but better add a new field in snapshot meta to make sure other peers won't apply it by mistake.

### CODE REVIEW

**Comment by BusyJay** [comment]
How can old version peer detect such error?

**Comment by Connor1996** [comment]
> How can old version peer detect such error?

Witness is enabled only when the cluster version meets the feature needs, so old version peer won't encounter snapshot generated for witness

**Comment by disksing** [comment]
PTAL @BusyJay

File: proto/raft_serverpb.proto [hunk 1]
```diff
@@ -45,6 +45,8 @@ message SnapshotMeta {
     repeated SnapshotCFFile cf_files = 1;
     // true means this snapshot is triggered for load balance
     bool for_balance = 2;
+    // true means this is an empty snapshot for witness
```
Comment (BusyJay) [inline proto/raft_serverpb.proto:48]: Is this still necessary after using file less snapshot?

Comment (Connor1996) [reply proto/raft_serverpb.proto:48]: yes, need it to decide how to handle the snapshot

**Review by BusyJay — APPROVED** [review]

**Pull request merged**